### PR TITLE
[ADF-4227] Sidenav Layout - support direction

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

preview

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
It looks a good idea. The only problem in the demo shell we already have a way to change the direction through the configuration editor maybe is better have a quick chat about it tomorrow to try merge those two

**Comment by DenysVuika** [comment]
@eromano @pionnegru Having a separate service looks like a good idea. That may help notifying various components to switch to RTL mode. The "adf-icon" (that wraps "mat-icon") definitely needs extra work too, as some Icons need to be altered, see https://material.angular.io/components/icon/overview#bidirectionality

File: demo-shell/src/app/app.routes.ts [hunk 1]
```diff
@@ -85,66 +85,61 @@ export const appRoutes: Routes = [
...
-        path: 'breadcrumb',
-        canActivate: [AuthGuardEcm],
-        component: AppLayoutComponent,
-        loadChildren: 'app/components/breadcrumb-demo/breadcrumb-demo.module#AppBreadcrumbModule'
-    },
-    {
-        path: 'notifications',
+        path: '',
         component: AppLayoutComponent,
+        data: { reuse: true },
```
Comment (DenysVuika) [inline demo-shell/src/app/app.routes.ts]: this thing does not exist in the ADF

**Comment by eromano** [comment]
I have add the data: { reuse: true } to avoid continue to reload of the appLayout
use the configuration editor to change rtl ltr instead of the select box
documentation

**Comment by eromano** [comment]
Screenshot 2019-04-10 at 18 20 05

File: demo-shell/src/app/app.routes.ts (hunk 1 above)
Comment (eromano) [reply demo-shell/src/app/app.routes.ts]: Passing a constant data stop the rerendering of the applayout I am investigating it

Comment (eromano) [reply demo-shell/src/app/app.routes.ts]: you are right is enough moving all under the right parent as I did reuse is a thing of my immagination :)

**Comment by pionnegru** [comment]
@eromano how do we target children components that need some css changes based on direction?

**Comment by eromano** [comment]
There are two failing unit test after some of our changes : 
AILED TESTS:
  SidenavLayoutComponent
    Template transclusion
      adf-sidenav-layout-header
        ✖ should contain the transcluded header template outside of the layout-container
        ✖ should call through the layout container's toggleMenu method

**Comment by pionnegru** [comment]
@cristinaj are these failing e2e valid for this PR?

**Comment by eromano** [comment]
Nope development doesn't have it red

**Comment by pionnegru** [comment]
so I guess this are random? https://travis-ci.org/Alfresco/alfresco-ng2-components/jobs/523587764#L1811

**Comment by eromano** [comment]
Not sure, development is green now. I rebase your PR let's see

**Pull request merged**